app/models/locker_application.rb

```diff
@@ -3,20 +3,31 @@
 class LockerApplication < ApplicationRecord
   belongs_to :user
   has_one :locker_assignment
+  belongs_to :building
 
   delegate :uid, :email, :name, :department, :status, to: :user
 
   def self.awaiting_assignment
     left_joins(:locker_assignment).where('locker_assignments.id is null').order('locker_applications.created_at')
   end
 
+  def self.mark_applications_complete
+    LockerApplication.where(complete: nil).find_each { |application| application.update(complete: true) }
+  end
+
   after_initialize do |_locker_application|
     if user.present?
       self.department_at_application ||= department
       self.status_at_application ||= status
     end
   end
 
+  def building_choices
```

Comment:
Would this method make more sense in the `Building` model?  I could go either way.